src/arch/x86_64.rs

```diff
@@ -72,10 +72,39 @@ pub extern "C" fn _start(info: &'static bootinfo::BootInfo) -> ! {
     mycelium_kernel::kernel_main(&bootinfo);
 }
 
-#[panic_handler]
-#[cfg(target_os = "none")]
-fn panic(info: &core::panic::PanicInfo) -> ! {
+pub(crate) fn oops(cause: &dyn core::fmt::Display) -> ! {
+    use core::fmt::Write;
+
+    unsafe { asm!("cli" :::: "volatile") }
```

Comment:
Should we also stub out the IDT to prevent NMIs from being run?

Reply:
alternate very cursed idea: swap IDTs so irq0 is the real oops handler, then do an `int 0` to raise the interrupt, thereby blocking any other NMI until the handler returns

Reply:
> Should we also stub out the IDT to prevent NMIs from being run?

hmm, if we just make the NMI handler "not present", we'll fault. i think we would either need a nop ISR, or do ixi's horrible thing.

Reply:
i think if an NMI happens during an oops, that's real bad news indeed...

Reply:
I think either ixi's idea or installing a nop-ISR are the main options here.